### ISSUE TITLE
authorized_keys assumes home directory is already created.

### ISSUE THREAD

**Issue by blinkus** [ansible/ansible-modules-core#59]
##### ISSUE TYPE

Feature Idea
##### COMPONENT NAME

authorized_key module
##### ANSIBLE VERSION

N/A
##### SUMMARY

The authorized_keys.py module uses the "os.mkdir()" function to create the .ssh directory if it doesn't exist.
It's possible that the users home directory hasn't been created. 
So using the "os.mkdirs()" function would recursively create the directory if the users home directory didn't exist.

Here is an error I came across when deploying ssh keys
  File "//.ansible/tmp/ansible-tmp-1412053423.23-52811687695291/authorized_key", line 175, in keyfile
    os.mkdir(sshdir, 0700)
OSError: [Errno 2] No such file or directory: '/users/lincolns/.ssh'

**Comment by lfrancke** [comment]
See this issue for more details: https://github.com/ansible/ansible/issues/7333

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by parberge** [comment]
+1 to fix this. Ran into this issue with following ansible version:

```
$ ansible --version
ansible 1.9.4
  configured module search path = /usr/share/ansible:plugins
```

**Comment by abadger** [comment]
We discussed this at last week's meeting and decided that this was not a feature we desired.  Creating an authorized_key for a non-existent home dir is more usefully expressed as an error to catch typos in usernames and other logic mistakes (for instance trying to upload keys for users who do not exist).  If creating the upper level directories is desired, it's easy to do it with a task which ensures that the directory is present before the authorized_keys task.
